libs/sdk-core/src/persist/sync.rs

```diff
@@ -170,7 +170,8 @@ impl SqliteStorage {
           lnurl_metadata,
           lnurl_withdraw_endpoint,
           attempted_amount_msat,
-          attempted_error
+          attempted_error,
+          external_metadata
          FROM remote_sync.payments_external_info
          WHERE payment_id NOT IN (SELECT payment_id FROM sync.payments_external_info);",
```

Comment:
Syncing metadata is a bit different since we allow updating it in contrary to other columns that are created once and immutable.
This creates a problematic scenario when one device sets the metadata and after sync to a remote one subsequent updates won't be synced again (since the payment already exists).
One way to solve this is use the updated_at timestamp to sync the same way we do with the swap fees (`sync_swaps_fees_local`)